File: term2/week1/week1-c/exercise10.c
#include <stdio.h>

int fib(n) {
    if (n <= 1) {
        return n;
    };
    return fib(n - 1) + fib(n - 2);
}

int main() {
    for (int i = 0; i < 42; i++)
    {
        int res = fib(i);
        printf("%i\n", res);
    }
    return 0;
}

File: term2/week1/week1-rs/exercise10.rs
fn main() {

    let mut i = 0;

    while i < 10 {
        let res = fib(i);
        println!("{}", res);
        i = i + 1;
    }
}

pub fn fib(n: i32) -> i32 {
    if n <= 1 {
        return n;
    }
    return fib(n-1) + fib(n-2);
}